### PR TITLE
source Send to MQTT

### PR DESCRIPTION
**Pull request by udo1toni**
This source adds send to MQTT functionality. 

Prequerities are an installed mosquitto client mosquitto_pub, which is called through the script.

### CODE REVIEW

**Comment by lgaetz** [comment]
Hello @udo1toni 

Great stuff, thanks for the contribution. We will require a code Contributor License Agreement from you in order to accept, do we have one on file already? If not, please see:
https://wiki.freepbx.org/pages/viewpage.action?pageId=28180643

**Comment by udo1toni** [comment]
Done :)

File: sources/source-Send_to_MQTT.module [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        if (($run_param['MQTT_broker'] != '') && ($run_param['MQTT_topic'] != '') && ($first_caller_id != '')) {
+            $port = ' -p '. $run_param['MQTT_port'];
+            $broker = ' -h '. $run_param['MQTT_broker'];
+            $user = "";
+            if($run_param['MQTT_user'] != ''){
+                $user = ' -u '. $run_param['MQTT_user'];
+            };
+            $password = "";
+            if($run_param['MQTT_user'] != ''){
```
Comment (lgaetz) [inline sources/source-Send_to_MQTT.module]: Hi @udo1toni  - I can't test, just scanning thru the code. Is this line correct, check to see if user is "" and then set password?

File: sources/source-Send_to_MQTT.module [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ * Version History
+ *     2020-10-14  Initial creation by hartmann1970@gmx.de
+ *     2020-10-15 added user, password, clientID by hartmann1970@gmx.de
+ *
+ *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***/
+
+class Send_to_MQTT extends superfecta_base {
+
+    public $description = "This source will send CID number and name to an MQTT topic for external processing. The MQTT topic is added with DID and payload is JSON format. To use this feature, mosqitto has to be installed.";
```
Comment (lgaetz) [inline sources/source-Send_to_MQTT.module]: mosquitto is misspelled in the description. This is an easy note to miss, and mosquitto is not installed by default in the FreePBX Distro. Would be good to test for it being installed and DebugPrint a warning that it's required.

File: sources/source-Send_to_MQTT.module (hunk 1 above)
Comment (udo1toni) [reply sources/source-Send_to_MQTT.module]: copy/paste error :) clearly it should be MQTT_password...

File: sources/source-Send_to_MQTT.module (hunk 2 above)
Comment (udo1toni) [reply sources/source-Send_to_MQTT.module]: Thanks. Missed that typo...

**Comment by lgaetz** [comment]
Hi @udo1toni 

I was supposed to write you back and ask that you please raise this PR with the project at git.freepbx.org, and forgot to do so. Once your user account at git.freepbx.org is properly associated with the CLA on file, we can accept the PR.

**Comment by udo1toni** [comment]
Hm, I'm a bit puzzled, as I already did send the Licence agreement months ago...

**Comment by creslin287** [comment]
Hey,

First off, thanks so much for the contribution.

Currently, github for FreePBX is a read only mirror of the code at git.freepbx.org. All code reviews are funneled through that repo, and so you'll need to resubmit there instead for this to move forward.

Best wishes,
Matthew Fredrickson

**Pull request closed without merging**